pkg/apis/core/validation/validation.go

```diff
@@ -2423,6 +2427,16 @@ func validateObjectFieldSelector(fs *core.ObjectFieldSelector, expressions *sets
 	return allErrs
 }
 
+func validateDownwardAPIHostIPs(fieldSel *core.ObjectFieldSelector, fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if !opts.AllowHostIPsField {
+		if fieldSel.FieldPath == "status.hostIPs" {
+			allErrs = append(allErrs, field.Invalid(fldPath, "status.hostIPs", "not allowed when feature gate 'PodHostIPs' is not enabled"))
+		}
+	}
+	return allErrs
+}
+
```

Comment:
https://github.com/kubernetes/kubernetes/pull/101566#discussion_r830937885
https://github.com/kubernetes/kubernetes/pull/101566#discussion_r838719070
https://github.com/kubernetes/kubernetes/pull/101566#discussion_r837686885

Turn on this feature and has a Pod that uses this feature then turn off this feature, here needs to be allowed.

``` yaml
apiVersion: v1
kind: Pod
metadata:
  name: test-pod
spec:
  containers:
    - name: test-container
      image: gcr.io/google_containers/busybox
      command: [ "/bin/sh", "-c", "env" ]
      env:
        - name: MY_POD_HOST_IPS
          valueFrom:
            fieldRef:
              fieldPath: status.hostIPs
  restartPolicy: Never
```
There is already such a pod running, then turning off this feature affects the existing pod.